### PR TITLE
add span per connection attempt

### PR DESCRIPTION
**Pull request by Tantalor93**
Signed-off-by: Ondrej Benkovsky <ondrej.benkovsky@wandera.com>

Hello, I would like to improve the observability of fanout plugin by adding opentracing spans per upstream connection attempt similarly as [CoreDNS forward plugin does](https://github.com/coredns/coredns/blob/master/plugin/forward/forward.go#L123). You can then better see the performance of each DNS upstream nameserver 

**Example how it looks like**:
Let's have a CoreDNS instance running with Corefile
```
.:53 {
    fanout . 8.8.8.8 9.9.9.9
    trace zipkin 127.0.0.1:9411 {
              service dns-service
    }
}
```
and locally running Zipkin in docker
```
docker run -d -p 9411:9411 openzipkin/zipkin
```

when you do the DNS request through this CoreDNS instance
```
dig @127.0.0.1 example.com
```

it will create a trace such as this (screen from Zipkin UI)

image

### CODE REVIEW

**Review by denis-tingaikin — COMMENTED** [review]
Thanks, @Tantalor93 for the patch! 
It makes sense. 
Please take a at my comment.

File: client.go [hunk 1]
```diff
@@ -64,6 +66,12 @@ func (c *client) Endpoint() string {
 
 // Request sends request to DNS server
 func (c *client) Request(ctx context.Context, r *request.Request) (*dns.Msg, error) {
+	span := ot.SpanFromContext(ctx)
+	if span != nil {
+		childSpan := span.Tracer().StartSpan("connect", ot.ChildOf(span.Context()))
```
[the remark this answers is not in the document]
Comment (Tantalor93) [reply client.go]: hello, thanks for commment 👍 

I tried to adjust it as you suggested, with one minor adjustment, I kept the `peer.address` tag, as it is one of the standard span tags according to the [opentracing convention](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table)

Regarding whether to tag `request` or `connect` span, I tagged the `request` span so it is earlier in the trace seen which nameserver it is connecting, but I can also add the tag to the `connect` span, what would you prefer?

[the remark this answers is not in the document]
Comment (Tantalor93) [reply client.go]: now it looks like this 
image

**Review by denis-tingaikin — APPROVED** [review]
LGTM

**Pull request merged**